datafusion/expr/src/expr_schema.rs

```diff
@@ -39,6 +40,13 @@ pub trait ExprSchemable {
 
     /// cast to a type with respect to a schema
     fn cast_to<S: ExprSchema>(self, cast_to_type: &DataType, schema: &S) -> Result<Expr>;
+
+    /// promote to a type with respect to a schema
```

Comment:
Can you please add some comments to clarify what the difference between "promote" and "cast" is?